vim/autoload/himalaya/mbox.lua

```diff
@@ -3,11 +3,30 @@ local action_state = require('telescope.actions.state')
 local finders = require('telescope.finders')
 local pickers = require('telescope.pickers')
 local sorters = require('telescope.sorters')
+local previewers = require('telescope.previewers')
 
 function mbox_picker(mboxes)
   pickers.new {
     results_title = 'Mailboxes',
-    finder = finders.new_table(mboxes),
+    finder = finders.new_table {
+        results = mboxes,
+        entry_maker = function(entry)
+            return {
+                value = entry,
+                display = entry,
+                ordinal = entry,
+                preview_command = function(entry, bufnr)
+                    vim.api.nvim_buf_call(bufnr, function()
+                        local success, _ = pcall(vim.fn['himalaya#mbox#post_input'](entry.value))
+                        if not (success) then
+                            -- TODO does not work since buffer is not modifiable
```

Comment:
Seems to be the correct approach for me. Maybe you can change the `modifiable` this way: `vim.bo.modifiable = true` ([sources](https://github.com/nanotee/nvim-lua-guide#using-meta-accessors)), and replace the buffer with the error returned by `himalaya#mbox#post_input` (or whatever).

Reply:
Good point @soywod! I also noticed that I had made a mistake of actually calling the function instead of giving it to `pcall` so fixed that :)

However, it doesn't seem to really work. Not sure if errors are properly propagated up to lua here. I.e. the error is created by calling `echoerr` I assume which perhaps does not get treated as an error by `pcall` but simply by vim.

Reply:
Indeed, all top-level functions are wrapped in a `try` block
https://github.com/soywod/himalaya/blob/0f1c490dad7f1df1b2cad5091ccdb8d66b26eb25/vim/autoload/himalaya/msg.vim#L21-L41
What can be done is to call directly the CLI from lua, either by calling the vim fn `himalaya#shared#cli#call` or by spawning a command. What do you think?

Reply:
@soywod Would it be okay to throw the error in the underlying vimscript-function? See latest commit. This makes the preview work but not sure if it breaks something else. What's the reason for catching any error?

If this does not work we could indeed directly call the cli, however it would give some duplicated code in the vimscript and lua files I guess. Which is maybe fine but would be nice to avoid.